File: examples/07_render_texture/asset/script/main.lua
local renderTexture = sn.RenderTexture.new()
local out = sn.Texture.new()
out:load("logo.png")
local sx = out:size().x
local sy = out:size().y
renderTexture:create(math.floor(sx), math.floor(sy))

local model = sn.Model.new()
model:load("DamagedHelmet.glb")

local transform = sn.Transform.new()
transform.rotation = sn.Vec3.new(90, 0, 0)

local pos = sn.Vec3.new(1, 1, 3)

function Update()
    if sn.Keyboard.isPressed(sn.Keyboard.ESCAPE) then
        sn.Script.load("main", ".")
    end
    sn.Graphics.getCamera():lookat(pos, sn.Vec3.new(0), sn.Vec3.new(0, 1, 0))
end

function Draw()
    sn.Graphics.bindPipeline(sn.BuiltinPipelines.get3D())
    sn.Graphics.setRenderTarget(renderTexture)
    sn.Graphics.drawModel(model, transform, model:getMaterial())
    -- Draw texture
    sn.Graphics.flush()
    sn.Graphics.readbackTexture(renderTexture, out)
    sn.Graphics.bindPipeline(sn.BuiltinPipelines.get2D())
    sn.Graphics.drawImage(out, sn.Rect.new(0, 0, sx, sy))
end

File: examples/07_render_texture/asset/script/main.py
import sinen as sn

texture = sn.Texture()
texture.fill(sn.Color(1))
renderTexture = sn.RenderTexture()
out = sn.Texture()
out.load("logo.png")
sx = out.size().x
sy = out.size().y
renderTexture.create(int(sx), int(sy))

model = sn.Model()
model.load("Suzanne.gltf")
draw3d = sn.Draw3D(texture)
draw3d.position = sn.Vec3(0)
draw3d.model = model

pos = sn.Vec3(1, 1, 3)


def update():
    pass
    if sn.Keyboard.is_pressed(sn.Keyboard.ESCAPE):
        sn.Script.load("main", ".")
    sn.Graphics.get_camera().lookat(pos, sn.Vec3(0), sn.Vec3(0, 1, 0))


def draw():
    sn.Graphics.set_render_target(renderTexture)
    sn.Graphics.bind_pipeline(sn.BuiltinPipelines.get_3d())
    sn.Graphics.draw3d(draw3d)
    # Draw texture
    sn.Graphics.flush()
    sn.Graphics.readback_texture(renderTexture, out)
    sn.Graphics.bind_pipeline(sn.BuiltinPipelines.get_2d())
    sn.Graphics.draw_image(out, sn.Rect(0, 0, sx, sy))
